doc/application_workflow/private_registry.md

```diff
@@ -0,0 +1,102 @@
+# Running a private registry with VIC
+
+In this example, we will run a private Docker registry on VIC and push and pull images using VIC.  VMware also offers an enterprise ready registry named [Harbor](https://github.com/vmware/harbor) that can be used in place of the base Docker registry.
+
+## Prerequisite
+
+Before going through this example, we need to re-emphasize some concepts around Docker and VIC.  The following examples are shown using Linux.  For Windows and Mac users, these examples should not differ much.
+
+Installing VIC will also require installing Docker locally.  When Docker is installed, we will get both a client (CLI or command line interface) and a daemon that handles all local container operations.  Local containers are those that run on the user local machine instead of a VMWare vSphere/ESXi environment.  The CLI is important as it will be most user's touchpoint for working with containers on VIC and on their local system.  The distinction between using the CLI against the two environment is very important in this example.  By default, the CLI will use the local Docker daemon.  After setting some environment variables, the CLI can be instructed to send all operations to VIC instead of the local Docker daemon.  The two environment variable are DOCKER_HOST and DOCKER_API_VERSION.
+
+In this example, we are deploying an insecure registry with no authentication for simplicity.  We will also be targeting an ESXi environment.
+
+## Workflow
+
+In terminal #1: (local Docker)
+
+1. Open a terminal and make sure that it will use the local Docker daemon.  At a command prompt, issue  
+    ```
+        $> unset DOCKER_HOST
+    ```  
+2. Install a VCH for a private registry using vic-machine
+3. Run [Docker's registry](https://docs.docker.com/registry/) on the first VCH
+4. Install a second VCH for running applications, making sure to specify --insecure-registry to ensure this second VIC can pull images from the insecure registry in the first VCH.
+5. At a terminal command prompt, using regular Docker, tag the images to be destined for the registry.
+6. Modify the docker systemd config file to allow pushing to an insecure registry
+7. Restart the docker daemon
+8. Push the image using the full tagged name (including host IP and port)
+
+In terminal #2: (VIC VCH)
+
+1. Open a terminal and make sure it is using the second VCH.  At a command prompt, issue  
+    ```
+        $> export DOCKER_HOST=tcp://<VCH_IP>:<VCH_PORT>
+        $> export DOCKER_API_VERSION=1.23
+    ```
+2. Pull the image from the registry VCH
+
+### Example run
+
+terminal 1:
+```
+$> unset DOCKER_HOST
+$> ./vic-machine-linux create --target=192.168.218.207 --image-store=datastore1 --name=vic-registry --user=root --password=vagrant --compute-resource="/ha-datacenter/host/esxbox.localdomain/Resources" --bridge-network=vic-network --no-tls --volume-store=datastore1/registry:default --force
```

Comment:
Suggest you update this to use --client-network-ip=fqdn, or to allow the registry container to be attached directly to a vsphere network with a static IP.
Either of those approaches will provide a stable registry endpoint for people to address.